openquake/commonlib/source.py

```diff
@@ -454,24 +474,20 @@ def get_rlzs_assoc(self, get_weight=lambda tm: tm.num_ruptures):
             else:  # full enumeration
                 rlzs = logictree.get_effective_rlzs(smodel.gsim_lt)
             if rlzs:
-                logging.info('Creating %d realization(s) for model '
-                             '%s, %s', len(rlzs), smodel.name, smodel.path)
                 idx = assoc._add_realizations(idx, smodel, rlzs)
-
-        if assoc.realizations:
-            if num_samples:
-                assert len(assoc.realizations) == num_samples
-                for rlz in assoc.realizations:
-                    rlz.weight = 1. / num_samples
-            else:
-                tot_weight = sum(rlz.weight for rlz in assoc.realizations)
-                if tot_weight == 0:
-                    raise ValueError('All realizations have zero weight??')
-                elif tot_weight < 1:
-                    logging.warn('Some source models are not contributing, '
-                                 'weights are being rescaled')
-                for rlz in assoc.realizations:
-                    rlz.weight = rlz.weight / tot_weight
+        if num_samples:
```

Comment:
Please check if there's actually any difference between the current situation and the old one